pkg/storage/fs/owncloud/owncloud.go

```diff
@@ -682,6 +684,14 @@ func (fs *ocfs) convertToResourceInfo(ctx context.Context, fi os.FileInfo, ip st
 
 	ri.PermissionSet = fs.permissionSet(ctx, ri.Owner)
 
+	// checksums
+	if _, ok := mdKeysMap[checksumsKey]; (!fi.IsDir()) && returnAllKeys || ok {
```

Comment:
I like to vote for more explicit error handling here. The handling of the IsDir() case should be done explicitely as that will ease debugging later big times. Also, would'nt a log message be helpful in the case that something is a dir that should not be or vise versa?

Also, I am only 38 years in programming and _still_ do not know quickly what `A && B || C` really evaluates to, how about helping me poor soul with additional parenthesis?